core/embed/projects/unix/main.c

```diff
@@ -498,6 +504,21 @@ static int sdl_event_filter(void *userdata, SDL_Event *event) {
   return 1;
 }
 
+void drivers_init() {
+#ifdef USE_TROPIC
+  uint8_t tropic_secret[SECRET_TROPIC_KEY_LEN] = {0};
+  secbool tropic_secret_ok = secret_tropic_get(tropic_secret);
+
+  tropic_init();
+  if (sectrue == tropic_secret_ok) {
+    if (tropic_handshake(tropic_secret) != TROPIC_SUCCESS) {
+      // ??
+    }
+  }
+  memzero(tropic_secret, sizeof(tropic_secret));
+#endif
```

Comment:
This should slightly differ from how it is done with optiga.

The difference is that you can communicate with an optiga without establishing a secure channel, but the set of commands you can use is restricted.

In the case of tropic, it doesn't make sense to continue if any of the functions `secret_tropic_get()`, `trocpi_init()` or `tropic_handshake` fails. So I would `ensure()` that all the functions succeed. And make sure to zero out the secret if any of the functions fails.

Reply:
Perfect, makes sense. I'll do that!